specs/merge/beacon-chain.md

```diff
@@ -81,13 +81,14 @@ class BeaconBlockBody(phase0.BeaconBlockBody):
 
 #### `BeaconState`
 
-*Note*: `BeaconState` fields remain unchanged other than addition of `application_state_root` and `application_block_hash`.
+*Note*: `BeaconState` fields remain unchanged other than addition of application payload fields.
 
 ```python
 class BeaconState(phase0.BeaconState):
     # Application-layer
     application_state_root: Bytes32  # [New in Merge]
     application_block_hash: Bytes32  # [New in Merge]
+    application_block_number: uint64 # [New in Merge]
```

Comment:
Would it better to define an `ApplicationState` container and bundle these in?

```python
class ApplicationState(Container): 
    root: Bytes32 
    block_hash: Bytes32
    block_number: uint64


class BeaconState(phase0.BeaconState):
    application_state: ApplicationState
```

Especially if we are going to add eip1559 fields later on

Reply:
I was thinking about this as well, but this is rather an `ApplicationBlockHeader` than `ApplicationState`. It would probably make sense to include all fields except for `transactions` into beacon state.

```python
class ApplicationBlockHeader(Container):
    block_hash: Bytes32
    parent_hash: Bytes32
    coinbase: Bytes20
    state_root: Bytes32
    number: uint64
    gas_limit: uint64
    gas_used: uint64
    receipt_root: Bytes32
    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]


class BeaconState(phase0.BeaconState)
    latest_application_block_header: ApplicationBlockHeader  # Akin to `latest_block_header`
```